src/JsonApiDotNetCore.MongoDb/MongoDbRepository.cs

```diff
@@ -153,10 +130,7 @@ private void AssertNoRelationship(TResource resourceFromRequest)
                 var rightResources = relationship.GetValue(resourceFromRequest);
                 if (rightResources != null)
                 {
-                    throw new JsonApiException(new Error(HttpStatusCode.BadRequest)
-                    {
-                        Title = "Relationships are not supported when using MongoDB."
-                    });
+                    throw new UnsupportedRelationshipException();
```

Comment:
It is sufficient to throw just when `_targetedFields.Relationships.Any()`

Better name for this method: AssertNoRelationshipsAreTargeted. And method parameter can be removed.